tests/test_monte_carlo.c

```diff
@@ -0,0 +1,155 @@
+#include <monte_carlo.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define BOARD_LENGTH 3
+#define BOARD_SIZE 8 // From 0 to 8
+
+void playAction(Board* board, Action* action) {
+  board->values[action->xPos * BOARD_LENGTH + action->yPos] = action->player;
+}
+
+static bool isValidAction(Board* board, Action* action, int player) {
+  if (action->player == player &&
+      board->values[action->xPos * 3 + action->yPos] == 0) {
+    return 1;
+  }
+  return 0;
+}
+
+void getPossibleActions(Board board, Action possibleActions[]) {
+  int nActions = 0;
+  int player = 1;
+  for (int x = 0; x < BOARD_LENGTH; ++x) {
+    for (int y = 0; y < BOARD_LENGTH; ++y) {
+      if (board.values[x * BOARD_LENGTH + y] == 0) {
+        for (int i = 0; i < board.nPlayers; ++i) {
+          possibleActions[nActions].player = player;
+          possibleActions[nActions].xPos = x;
+          possibleActions[nActions].yPos = y;
+          nActions++;
+          player = -player;
+        }
+      }
+    }
+  }
+}
+
+static int getNumPossibleActions(Board board) {
+  int nActions = 0;
+  for (int x = 0; x < BOARD_LENGTH; ++x) {
```

Comment:
Instead of using 2 for loops with 2 variables and computing the address each time, you should have 1 for loop that iterates from 0 to BOARD_LENGTH * BOARD_LENGTH

```
for(int i = 0; i < BOARD_LENGTH * BOARD_LENGTH; ++i)
{
    if(board.values[i] == 0)
    {
        ++nActions;
    }
}
```